middleware/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/miekg/coredns/middleware/proxy"
+//	"github.com/miekg/coredns/middleware/singleflight"
+
+    "github.com/miekg/dns"
+	"golang.org/x/net/context"
+)
+
+type Kubernetes struct {
+	Next       middleware.Handler
+	Zones      []string
+	Proxy      proxy.Proxy // Proxy for looking up names during the resolution process
+	Ctx        context.Context
+//	Inflight   *singleflight.Group
+    APIConn    *k8sc.K8sConnector
+}
+
+
+func (g Kubernetes) getZoneForName(name string) (string, []string) {
+    var zone string
+    var segments []string
+
+    for _, z := range g.Zones {
+        if dns.IsSubDomain(z, name) {
+            zone = z 
+    
+            segments = dns.SplitDomainName(name)
+            segments = segments[:len(segments) - dns.CountLabel(zone)]
+        }
+    }   
+
+    return zone, segments
+} 
+
+
+// Records looks up services in kubernetes.
+// If exact is true, it will lookup just
+// this name. This is used when find matches when completing SRV lookups
+// for instance.
+func (g Kubernetes) Records(name string, exact bool) ([]msg.Service, error) {
+
+    zone, serviceSegments := g.getZoneForName(name)
+
+    var serviceName string
+    var namespace string
+
+    // For initial implementation, assume namespace is first serviceSegment
+    // and service name is remaining segments.
+    serviceSegLen := len(serviceSegments)
+    if serviceSegLen >= 2 {
+        namespace = serviceSegments[serviceSegLen-1]
+        serviceName = strings.Join(serviceSegments[:serviceSegLen-1], ".")
+    }
+    // else we are looking up the zone. So handle the NS, SOA records etc.
+
+    fmt.Println("[debug] zone: ", zone)
+    fmt.Println("[debug] servicename: ", serviceName)
+    fmt.Println("[debug] namespace: ", namespace)
+    fmt.Println("[debug] APIconn: ", g.APIConn)
+
+    k8sItem := g.APIConn.GetServiceItemInNamespace(namespace, serviceName)
+    fmt.Println("[debug] k8s item:", k8sItem)
+
+    switch {
+        case exact && k8sItem == nil:
+            fmt.Println("here2")
+            return nil, nil
+    }
+
+    if k8sItem == nil {
+        // Did not find item in k8s
+        return nil, nil
+    }
+
+    fmt.Println("[debug] clusterIP:", k8sItem.Spec.ClusterIP)
+
+    for _, p := range k8sItem.Spec.Ports {
+        fmt.Println("[debug]    host:", name)
+        fmt.Println("[debug]    port:", p.Port)
+    }
+
+    clusterIP := k8sItem.Spec.ClusterIP
+    var records []msg.Service
+    for _, p := range k8sItem.Spec.Ports{
+        s := msg.Service{Host: clusterIP, Port: p.Port}
+        records = append(records, s)
+    }
+
+    return records, nil
+}
+
+/*
+// Get performs the call to the Kubernetes http API.
+func (g Kubernetes) Get(path string, recursive bool) (bool, error) {
+
+    fmt.Println("[debug] in Get path: ", path)
+    fmt.Println("[debug] in Get recursive: ", recursive)
+
+	return false, nil
+}
+*/
+
+func (self Kubernetes) splitDNSName(name string) []string {
```

Comment:
You prefer "this"? ;)
Changed to match '(g Kubernetes)' usage in this file. In the k8sclient.go file, changed to use '(c *K8sConnector)'.